src/python/pants/util/meta.py

```diff
@@ -6,6 +6,13 @@
 from typing import Any, Callable, Optional, Type, TypeVar, Union
 
 
+T = TypeVar("T")
+C = TypeVar("C")
```

Comment:
What do these variables mean?

Reply:
`T` is meant to be a general type. This is a common convention for MyPy.

`C` is short for class. Originally I tried setting `bound=Type` but couldn't get it to work. Your review prompted me to try again and I got it working! It also allows us to remove 3 type ignores! Thanks!